middleware/response.go

```diff
@@ -23,27 +29,42 @@ type badResponseLoggingWriter struct {
 	writeError    error // The error returned when downstream Write() fails.
 }
 
-// newBadResponseLoggingWriter makes a new badResponseLoggingWriter.
-func newBadResponseLoggingWriter(rw http.ResponseWriter, buffer io.Writer) *badResponseLoggingWriter {
-	return &badResponseLoggingWriter{
+// flushingBadResponseLoggingWriter is a badResponseLoggingWriter that
+// implements http.Flusher.
+type flushingBadResponseLoggingWriter struct {
+	nonFlushingBadResponseLoggingWriter
+	f http.Flusher
+}
+
+// newBadResponseLoggingWriter makes a new badResponseLoggingWriter or
```

Comment:
This comment is slightly incorrect - the type is always `badResponseLoggingWriter` but concretely two other types.  But the concrete type is not part of the observable behaviour so not necessary to mention here.

Reply:
Good point--I'll clean that up!